File: archive/legacy-review/gmx-services/gmx-common-partners/gmx-common-partners/partner-sbtech/src/main/java/net/flipsports/gmx/common/internal/partner/sbtech/cons/SBTechSportType.java
package net.flipsports.gmx.common.internal.partner.sbtech.cons;

import net.flipsports.gmx.common.internal.partner.commons.cons.ConsMapping;
import net.flipsports.gmx.common.internal.partner.commons.cons.SportType;

public enum SBTechSportType {
  SOCCER(1L, SportType.SOCCER),

  FOOTBALL(3L, SportType.FOOTBALL),

  HORSE_RACING(61L, SportType.HORSE_RACING),

  GREYHOUNDS(66L, SportType.GREYHOUNDS),

  ENHANCED_ODDS(79L, SportType.ENHANCED_ODDS);

  private final long sbtechId;
  private final SportType sportType;

  SBTechSportType(long sbtechId, SportType sportType) {
    this.sbtechId = sbtechId;
    this.sportType = sportType;
  }

  public long getSbtechId() {
    return sbtechId;
  }

  public SportType getSportType() {
    return sportType;
  }

  public static final ConsMapping<Long, SBTechSportType> MAPPING =
      new ConsMapping<>(SBTechSportType.values(), SBTechSportType::getSbtechId);

}

File: archive/legacy-review/gmx-services/gmx-site-extensions/gmx-site-extensions/data-feed/src/main/scala/gmx/common/internal/partner/sbtech/cons/SBTechSportType.scala
package gmx.common.internal.partner.sbtech.cons

import scala.collection.breakOut

import enumeratum.Enum
import enumeratum.EnumEntry
import tech.argyll.video.domain.model.SportType

sealed trait SBTechSportType extends EnumEntry {
  def sbtechId: String
  def sportType: SportType
}

object SBTechSportType extends Enum[SBTechSportType] {
  class SBTechSportTypeRecord(val sbtechId: String, val sportType: SportType) extends SBTechSportType

  val values = findValues

  private val byId: Map[String, SBTechSportType] = values.map(elem => (elem.sbtechId, elem))(breakOut)

  final case object Soccer extends SBTechSportTypeRecord("1", SportType.SOCCER)
  final case object Football extends SBTechSportTypeRecord("3", SportType.FOOTBALL)
  final case object HorseRacing extends SBTechSportTypeRecord("61", SportType.HORSE_RACING)
  final case object EnhancedOdds extends SBTechSportTypeRecord("79", SportType.ENHANCED_ODDS)

  def findById(in: String): Option[SBTechSportType] = byId.get(in)
}
